providers/universal-provider/src/UniversalProvider.ts

```diff
@@ -249,11 +254,25 @@ export class UniversalProvider implements IUniversalProvider {
       throw new Error("Sign Client not initialized");
     }
 
-    Object.keys(this.namespaces).forEach((namespace) => {
-      const accounts = this.session?.namespaces[namespace].accounts || [];
+    if (!this.session) {
+      throw new Error("Session not initialized. Please call connect() before enable()");
+    }
+
+    const providersToCreate = [
+      ...new Set(
+        Object.keys(this.session.namespaces).map((namespace) => parseNamespaceKey(namespace)),
+      ),
+    ];
+    providersToCreate.forEach((namespace) => {
+      if (!this.session) return;
```

Comment:
Saw the prior non null warning. Weird that this is still needed as an explicit check for TS to pick up that session is defined if you already have the `if (!this.session) ...` above?

Reply:
Yeah, seems redundant but at least ts is not raising warnings so no biggie